src/Foundatio/Caching/InMemoryCacheClient.cs

```diff
@@ -183,18 +183,23 @@ public Task<int> RemoveAllAsync(IEnumerable<string> keys = null)
 
     public Task<int> RemoveByPrefixAsync(string prefix)
     {
-        var keysToRemove = new List<string>();
         string normalizedPrefix = String.IsNullOrWhiteSpace(prefix) ? "*" : prefix.Trim();
-        var regex = new Regex(String.Concat("^", Regex.Escape(normalizedPrefix.Contains("*") ? normalizedPrefix : $"{normalizedPrefix}*"), "$").Replace("\\*", ".*?"));
+        if (String.Equals(normalizedPrefix, "*"))
+            return RemoveAllAsync();
+
+        var keysToRemove = new List<string>();
+
         try
         {
+            var regex = new Regex(String.Concat("^", Regex.Escape(normalizedPrefix.EndsWith("*") ? normalizedPrefix : $"{normalizedPrefix}*"), "$").Replace("\\*", ".*?"));
```

Comment:
moved this into the try so if it throws it's logged. Also changed the regex to endsWith so it better handles prefixes.